sasl.go

```diff
@@ -92,3 +97,117 @@ func ConnSASLAnonymous() ConnOption {
 		return nil
 	}
 }
+
+// ConnSASLXOAUTH2 enables SASL XOAUTH2 authentication for the connection.
+//
+// The saslMaxFrameSizeOverride parameter allows the limit that governs the maximum frame size this client will allow
+// itself to generate to be raised for the sasl-init frame only.  Set this when the size of the size of the SASL XOAUTH2
+// initial client response (which contains the username and bearer token) would otherwise breach the 512 byte min-max-frame-size
+// (http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-transport-v1.0-os.html#definition-MIN-MAX-FRAME-SIZE). Pass -1
+// to keep the default.
+//
+// SASL XOAUTH2 transmits the bearer in plain text and should only be used
+// on TLS/SSL enabled connection.
+func ConnSASLXOAUTH2(username, bearer string, saslMaxFrameSizeOverride uint32) ConnOption {
+	return func(c *conn) error {
+		// make handlers map if no other mechanism has
+		if c.saslHandlers == nil {
+			c.saslHandlers = make(map[symbol]stateFunc)
+		}
+
+		response, err := saslXOAUTH2InitialResponse(username, bearer)
+		if err != nil {
+			return err
+		}
+
+		// add the handler the the map
+		c.saslHandlers[saslMechanismXOAUTH2] = func() stateFunc {
+			handler := saslXOAUTH2Handler{
+				conn:                 c,
+				maxFrameSizeOverride: saslMaxFrameSizeOverride,
+				response:             response,
+			}
+			return handler.init()
+		}
```

Comment:
The anonymous function can be omitted by constructing `handler` before the map assignment and setting `c.saslHandlers[saslMechanismXOAUTH2] = handler.init`.

Reply:
Good point, much neater. Done.